chainer/utils/type_check.py

```diff
@@ -220,6 +220,22 @@ def _eval_expr(v):
         return v
 
 
+def _repr(v):
+    if isinstance(v, Expr):
+        return str(v)
+    elif isinstance(v, list):
+        return '[{0}]'.format(', '.join(map(_repr, v)))
+    elif isinstance(v, tuple):
+        if len(v) == 0:
+            return '()'
+        elif len(v) == 1:
+            return '({0},)'.format(_repr(v[0]))
+        else:
+            return '({0})'.format(', '.join(map(_repr, v)))
```

Comment:
How about delegating the stringization here to tuple.__str__ like `str(tuple(map(_repr, v)))`?